app/feature/feature-edit-coinsured/src/main/kotlin/com/hedvig/android/feature/editcoinsured/ui/InsuredRow.kt

```diff
@@ -0,0 +1,185 @@
+import com.hedvig.android.core.icons.hedvig.small.hedvig.Lock
+import com.hedvig.android.core.icons.hedvig.small.hedvig.Plus
+import com.hedvig.android.core.ui.text.HorizontalItemsWithMaximumSpaceTaken
+import hedvig.resources.R
+
+@Composable
+internal fun InsuredRow(
+  displayName: String?,
+  details: String?,
+  hasMissingInfo: Boolean,
+  allowEdit: Boolean,
+  isMember: Boolean,
+  onRemove: () -> Unit,
+  onEdit: () -> Unit,
+) {
+  HorizontalItemsWithMaximumSpaceTaken(
+    startSlot = {
+      Row(
+        verticalAlignment = Alignment.CenterVertically,
+        modifier = Modifier.padding(vertical = 16.dp),
+      ) {
+        Column {
+          if (displayName != null) {
+            Text(
+              text = displayName,
+              color = if (isMember) {
+                MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = 0.5f)
+              } else {
+                Color.Unspecified
+              },
+            )
+          } else {
+            Text(stringResource(id = R.string.CONTRACT_COINSURED))
+          }
+
+          if (details != null) {
+            Text(
+              text = details,
+              color = if (isMember) {
+                MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = 0.5f)
+              } else {
+                MaterialTheme.colorScheme.onSurfaceVariant
+              },
+            )
+          } else {
+            Text(
+              text = stringResource(id = R.string.CONTRACT_NO_INFORMATION),
+              color = MaterialTheme.colorScheme.onSurfaceVariant,
+            )
+          }
+        }
+      }
+    },
+    endSlot = {
+      Row(
+        horizontalArrangement = Arrangement.End,
+        verticalAlignment = Alignment.CenterVertically,
+        modifier = Modifier.padding(vertical = 16.dp),
+      ) {
+        when {
+          isMember -> {
+            Icon(
+              imageVector = HedvigIcons.Lock,
+              contentDescription = "Locked",
+              tint = MaterialTheme.colorScheme.onSurfaceVariant.copy(alpha = 0.5f),
+              modifier = Modifier.size(16.dp),
+            )
+          }
+
+          allowEdit && hasMissingInfo -> {
+            Text(stringResource(id = R.string.GENERAL_ADD_INFO_BUTTON))
+            Spacer(modifier = Modifier.width(6.dp))
+            Icon(
+              imageVector = HedvigIcons.Plus,
+              contentDescription = "Add",
+              modifier = Modifier.size(16.dp),
+            )
+          }
+
+          allowEdit -> {
+            Text(stringResource(id = R.string.CONTRACT_EDIT_INFO))
+          }
+
+          else -> {
+            Icon(
+              imageVector = HedvigIcons.X,
+              contentDescription = "Remove",
+              modifier = Modifier.size(16.dp),
+            )
+          }
+        }
+      }
+    },
+    spaceBetween = 8.dp,
+    modifier = Modifier
+      .clickable(enabled = !isMember) {
+        if (hasMissingInfo) {
```

Comment:
Should this also then check for if `allowEdit` is true?

Reply:
Will change